frontend/src/components/software/SoftwareDetailsContainer.jsx

```diff
@@ -2,9 +2,60 @@ import React, {Component} from 'react';
 import SoftwareDetailsComponent from './SoftwareDetailsComponent';
 
 class SoftwareDetailsContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.addCall = this.addCall.bind(this);
+    this.state = {
+      name: '',
+      key: '',
+      availableKeys: '',
+      duration: '',
+      mode: '',
+      id: '',
+    };
+  }
+
+
+
+  addCall(){
+    fetch('http://localhost:8080/api/software',{
+      method: 'POST',
+      mode: 'cors',
+      body: JSON.stringify(this.state),
+      headers: {
+        'Content-Type': 'application/json',
+        'Access-Control-Allow-Origin': '*',
+      }
+    }).then((response) => response.json())
+      .then((responseJson) => {
+        console.log(responseJson);
+        return responseJson;
```

Comment:
W tym miejscu trzeba zapisać dane w `this.state`, korzystając z metody `setState`. Ta metoda automatycznie wywoła metodę `render`.